src/org/sosy_lab/java_smt/solvers/cvc5/CVC5InterpolatingProver.java

```diff
@@ -0,0 +1,370 @@
+// This file is part of JavaSMT,
+// an API wrapper for a collection of SMT solvers:
+// https://github.com/sosy-lab/java-smt
+//
+// SPDX-FileCopyrightText: 2023 Dirk Beyer <https://www.sosy-lab.org>
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package org.sosy_lab.java_smt.solvers.cvc5;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Iterables;
+import io.github.cvc5.Kind;
+import io.github.cvc5.Solver;
+import io.github.cvc5.Term;
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.sosy_lab.common.ShutdownNotifier;
+import org.sosy_lab.java_smt.api.BooleanFormula;
+import org.sosy_lab.java_smt.api.FormulaManager;
+import org.sosy_lab.java_smt.api.InterpolatingProverEnvironment;
+import org.sosy_lab.java_smt.api.SolverContext.ProverOptions;
+import org.sosy_lab.java_smt.api.SolverException;
+
+public class CVC5InterpolatingProver extends CVC5AbstractProver<Term>
+    implements InterpolatingProverEnvironment<Term> {
+
+  private final FormulaManager mgr;
+  private final Set<ProverOptions> solverOptions;
+  private final int seed;
+
+  CVC5InterpolatingProver(
+      CVC5FormulaCreator pFormulaCreator,
+      ShutdownNotifier pShutdownNotifier,
+      @SuppressWarnings("unused") int randomSeed,
+      Set<ProverOptions> pOptions,
+      FormulaManager pMgr) {
+    super(pFormulaCreator, pShutdownNotifier, randomSeed, pOptions, pMgr);
+    mgr = pMgr;
+    solverOptions = pOptions;
+    seed = randomSeed;
+  }
+
+  /**
+   * Sets the same solver Options of the Original Solver to the separate solvertoSet, except for
+   * produce-interpolants which is set here. From CVC5AbstractProver Line 66
+   */
+  private void setSolverOptions(int randomSeed, Set<ProverOptions> pOptions, Solver solvertoSet) {
```

Comment:
Please do not just copy code, but correctly refactor it to be reusable.

In that case:
- Add a new parameter "pSolver" to the super-method and make the method "protected".
- Then simplify this method to only two lines of code:
```
@Override
protected void setSolverOptions(int randomSeed, Set<ProverOptions> pOptions, Solver pSolver) {
  super.setSolverOptions(randomSeed, pOptions, pSolver);
  solvertoSet.setOption("produce-interpolants", "true");
}
```

This is much easier to understand and avoids redundancy.

Reply:
I see how it is done now. Thanks.